src/hooks/cashu/useCashu.ts

```diff
@@ -262,9 +262,11 @@ export const useCashu = () => {
          lockBalance();
 
          await addProofs([...newProofs, ...change]);
-         if (opts.amount || opts.max) {
+         if ((opts.amount || opts.max) && !swappedToUnlock) {
```

Comment:
`swappedToUnlock` means that we got locked ecash and needed to swap it with the mint for unlocked ecash so that we can melt. 
`removeProofs` throws if you try to remove proofs that don't exist, and swapping results in new proofs that do not exist.

 This is not the ideal solution, but we are waiting for cashu-ts to add the ability to provide a private key directly to `meltTokens` so that the P2PK signature can be created when melting rather than doing it in two operations.